src/ErrorHandling/ErrorModal.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { Modal } from "../ui/React/Modal";
+import { Box, Button, Typography } from "@mui/material/";
+import { ErrorRecord, ErrorState } from "./ErrorState";
+import { Router } from "../ui/GameRoot";
+import { SimplePage } from "@enums";
+import { useRerender } from "../ui/React/hooks";
+import { OptionSwitch } from "../ui/React/OptionSwitch";
+import { LogBoxEvents } from "../ui/React/LogBoxManager";
+import { recentScripts } from "../Netscript/RecentScripts";
+
+export function ErrorModal(): React.ReactElement {
+  const { classes } = useStyles();
+  const rerender = useRerender();
+  const [error, setError] = useState<ErrorRecord | null>(ErrorState.ActiveError);
+
+  useEffect(() => {
+    const listener = (newError: ErrorRecord) => {
+      if (newError.force || (Router.page() !== SimplePage.ActiveScripts && !ErrorState.PreventModals)) {
+        setError(newError);
+        rerender();
+      }
+    };
+    ErrorState.ErrorUpdate.subscribe(listener);
```

Comment:
```suggestion
    return ErrorState.ErrorUpdate.subscribe(listener);
```
Forgetting to use the cleanup function can create memory leaks. It's not a problem here because `ErrorModal` is created in `GameRoot` and it's not unmounted, but it's a problem in `src\ErrorHandling\RecentErrorsPage.tsx`.